synapse/http/site.py

```diff
@@ -83,11 +91,11 @@ def __init__(self, channel, *args, max_request_body_size=1024, **kw):
         self._is_processing = False
 
         # the time when the asynchronous request handler completed its processing
-        self._processing_finished_time = None
+        self._processing_finished_time: Optional[float] = None
 
         # what time we finished sending the response to the client (or the connection
         # dropped)
-        self.finish_time = None
+        self.finish_time: Optional[float] = None
```

Comment:
Not sure why mypy started being unhappy, but it was interpreting these as being of type `NoneType` throughout the program, causing problems later.